### PR TITLE
Add configuration server to default_main

### PR DESCRIPTION
**Pull request by paf31**
This adds a configuration server to `default_main` which can be run using `cargo run -- configuration serve`:

```sh
> cargo run -- configuration serve
Starting server on 0.0.0.0:9100

> # Get a minimal configuration
> curl http://0.0.0.0:9100              
null

> # Elaborate a configuration
> curl http://0.0.0.0:9100 -X POST -H 'Content-Type: application/json' -d 'null'       
null

> # Validate configuration
> curl http://0.0.0.0:9100/validate -X POST -H 'Content-Type: application/json' -d 'null' | jq .
{
  "schema": {
    "scalar_types": {},
    "object_types": {},
    "collections": [],
    "functions": [],
    "procedures": []
  }
}

> # Get the JSON schema for the configuration
> curl http://0.0.0.0:9100/schema | jq .    
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Null",
  "type": "null"
}
```

### CODE REVIEW

File: src/default_main.rs [hunk 1]
```diff
@@ -49,6 +47,24 @@ struct ServeCommand {
...
+enum ConfigurationSubcommand {
+    #[command()]
+    Serve(ServeConfigurationCommand),
+}
+
+#[derive(Clone, Parser)]
+struct ServeConfigurationCommand {
+    #[arg(long, value_name = "PORT", env = "PORT")]
+    port: Option<String>,
+}
```
Comment (sordina) [inline src/default_main.rs:66]: Does this default to something?

File: src/default_main.rs [hunk 2]
```diff
@@ -241,14 +254,107 @@ async fn post_query<C: Connector>(
...
-    C::ConfigureArgs: Clone + Send + Sync + Args,
-    C::RawConfiguration: Serialize,
+    C::RawConfiguration: Serialize + DeserializeOwned + JsonSchema + Clone + Sync + Send,
+    C::Configuration: Sync + Send,
+{
+    match command.command {
+        ConfigurationSubcommand::Serve(serve_command) => {
+            serve_configuration::<C>(serve_command).await
+        }
+    }
```
Comment (sordina) [inline src/default_main.rs:268]: 👍

**Review by sordina — APPROVED** [review]
Looks great!

File: src/default_main.rs (hunk 1 above)
Comment (paf31) [reply src/default_main.rs:66]: Yes, 9100

**Pull request merged**